src/storage_common.c

```diff
@@ -72,11 +73,12 @@ static int thingset_storage_init(void)
         thingset_set_update_callback(&ts, TS_SUBSET_NVM, thingset_storage_update_handler);
     }
 
+#ifdef CONFIG_THINGSET_STORAGE_AUTOSAVE
     if (IS_ENABLED(CONFIG_THINGSET_STORAGE_AUTOSAVE)) {
         thingset_sdk_reschedule_work(&storage_work,
                                      K_HOURS(CONFIG_THINGSET_STORAGE_AUTOSAVE_INTERVAL));
     }
-
+#endif
```

Comment:
Same here. We should move to compile-time only check.

And please keep the empty line before the return statement as a separation.

Reply:
i was not sure if the runtime check has some sideeffects if i remove them. i remove them and add the empty line which went gone for an unknown reason.